force-app/main/default/classes/RD2_ETableController.cls

```diff
@@ -365,4 +365,32 @@ public with sharing class RD2_ETableController {
         
         return permissions.canRead(npe03__Recurring_Donation__c.SObjectType, fields);
     }
+
+    private without sharing class RDScheduleClass {
+        public Map<Id, List<RecurringDonationSchedule__c>> queryRows(Set<Id> recordIds) {
+            Map<Id, List<RecurringDonationSchedule__c>> mapRDSchedule = new Map<Id, List<RecurringDonationSchedule__c>>();
+            for(RecurringDonationSchedule__c schedule : [   SELECT Active__c, 
```

Comment:
This is going to fail in an org with more than 50K schedule records. The query must be filtered by either `RecurringDonation__c` or probably more usefully by `RecurringDonation__r.npe03__Contact__c`. In other words, this query must only retrieve the specific schedule records for the Donor (Contact) that it's being displayed for.